tests/lib/snaps/basic-binaries/bin/disallowed

```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+dd if=/dev/kmsg iflag=nonblock bs=512 count=1
```

Comment:
Instead of doing this, can we please use the `head` tool and call this file as "head", inside it doing simply:
```
head "$@"
```
So that inside the test we can do:
```
basic-binaries.head -1 /dev/kmsg

This makes it reusable and much easier to understand what the output is when it fails.

I think we should do the same for the "echo" and "cat" helpers. Right now they are quite surprising, because they're not actually echo and cat, but simply call these internally with a custom argument. cat makes pretty much no sense for example, as it cats itself.